ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -242,8 +318,445 @@ public void testValidDataSizeAfterDeletes() throws InterruptedException, StoreEx
   }
 
   /**
-   * Test the static method that converts the quota stats in the form of a nested {@link Map} to the generic
-   * {@link StatsSnapshot} object.
+   * Basic test to verify that the {@link BlobStoreStats} can scan the index, populate the buckets and use these buckets
+   * to report stats correctly.
+   * @throws StoreException
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void testBucketingBasic() throws StoreException, InterruptedException, IOException {
+    assumeTrue(bucketingEnabled);
+    final CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
+    long logSegmentForecastOffsetMs = state.time.milliseconds();
+    int bucketCount = 2 * (int) (logSegmentForecastOffsetMs / BUCKET_SPAN_IN_MS);
+    long expiresAtInMs = ((long) bucketCount - 2) * BUCKET_SPAN_IN_MS;
+    // add 3 puts with expiry
+    state.addPutEntries(3, CuratedLogIndexState.PUT_RECORD_SIZE, expiresAtInMs);
+    int expectedThrottleCount = state.referenceIndex.size();
```

Comment:
After you fixed the throttler to throttle based on no of index entries(instead of no index segments), did you run your tests. I guess expectedThrottleCount needs fixes here